lib/ansible/modules/storage/netapp/sf_volume_manager.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/python
+
+# (c) 2017, NetApp, Inc
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+
+module: sf_volume_manager
+
+short_description: Manage SolidFire volumes
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+description:
+- Create, destroy, or update volumes on SolidFire
+
+options:
+
+    state:
+        description:
+        - Whether the specified volume should exist or not.
+        required: true
+        choices: ['present', 'absent']
+
+    name:
+        description:
+        - The name of the volume to manage.
+        required: true
+
+    account_id:
+        description:
+        - Account ID for the owner of this volume.
+        required: true
+
+    512emulation:
```

Comment:
I didn't see such a argument in arg_spec, is this `enable512e`? Note you can make aliases in arg_specif you like to keep the logic.

See my comment in the arg spec.